enterprise/jmx-monitoring/src/main/java/io/crate/beans/QueryStats.java

```diff
@@ -21,53 +21,65 @@
 import com.google.common.base.Suppliers;
 import com.google.common.collect.ImmutableSet;
 import io.crate.execution.engine.collect.stats.JobsLogs;
-import io.crate.expression.reference.sys.job.JobContextLog;
+import io.crate.metadata.sys.ClassifiedMetrics.Metrics;
 import io.crate.planner.Plan.StatementType;
+import io.crate.planner.operators.StatementClassifier;
+import org.HdrHistogram.Histogram;
 
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Supplier;
 
-
 public class QueryStats implements QueryStatsMBean {
 
     private static final Set<StatementType> CLASSIFIED_STATEMENT_TYPES =
         ImmutableSet.of(StatementType.SELECT, StatementType.INSERT, StatementType.UPDATE, StatementType.DELETE);
 
     static class Metric {
 
-        private final long elapsedSinceUpdateInMs;
-
-        private long count;
+        private long measurementIntervalInMs;
+        private long totalCount;
         private long sumOfDurations;
 
-        Metric(long duration, long elapsedSinceUpdateInMs) {
-            this.elapsedSinceUpdateInMs = elapsedSinceUpdateInMs;
-            sumOfDurations = duration;
-            count = 1;
+        Metric(long sumOfDurations, long totalCount, long measurementIntervalInMs) {
+            this.measurementIntervalInMs = measurementIntervalInMs;
+            this.sumOfDurations = sumOfDurations;
+            this.totalCount = totalCount;
         }
 
-        void inc(long duration) {
-            sumOfDurations += duration;
-            count++;
+        void set(long sumOfDurations, long totalCount, long measurementIntervalInMs) {
+            this.measurementIntervalInMs = measurementIntervalInMs;
+            this.sumOfDurations = sumOfDurations;
+            this.totalCount = totalCount;
+        }
+
+        void incMetricsAndSetMeasurementInterval(long duration, long count, long measurementIntervalInMs) {
+            this.measurementIntervalInMs = measurementIntervalInMs;
+            this.sumOfDurations += duration;
+            this.totalCount += count;
         }
 
         double statementsPerSec() {
-            return count / (elapsedSinceUpdateInMs / 1000.0);
+            return totalCount / (measurementIntervalInMs / 1000.0);
         }
 
         double avgDurationInMs() {
-            return sumOfDurations / (double) count;
+            return sumOfDurations / (double) totalCount;
```

Comment:
The histograms already have an average. We could also use that value.

Reply:
The average in the histograms are since the node is started so we can't use them in the "since last read" measurements.